app/controllers/notifications_controller.rb

```diff
@@ -13,17 +13,19 @@ def index
     scope = scope.reason(params[:reason]) if params[:reason].present?
     scope = scope.type(params[:type]) if params[:type].present?
     scope = scope.status(params[:status]) if params[:status].present?
+    scope = scope.owned_by(current_user)
     @notifications = scope.newest
   end
 
   def archive
-    notification = Notification.find(params[:id])
-    notification.update_attributes(archived: true)
+    notification = Notification.find_by(id: params[:id], user: current_user)
+    notification.update_attributes(archived: true) if notification.present?
```

Comment:
If we use `find(params[:id])` in the line above then rails will raise an `NotFound` exception here if it's missing so no need to check for presence.